aiida/cmdline/utils/migration/v03_to_v04.py

```diff
@@ -0,0 +1,528 @@
+        if 'type' in value and value['type'].startswith('data.'):
+            value['type'] = value['type'].replace('data.', 'node.data.', 1)
+
+
+def migration_trajectory_symbols_to_attribute(data, folder):
+    """Apply migrations: 0026 - REV. 1.0.26 and 0027 - REV. 1.0.27
+    Create the symbols attribute from the repository array for all `TrajectoryData` nodes.
+    """
+    for node_id, content in data['export_data']['Node'].items():
+        if 'type' in content and content['type'] == 'node.data.array.trajectory.TrajectoryData.':
+            uuid = content['uuid']
+            symbols_path = os.path.join(
+                folder.get_abs_path('nodes'), uuid[0:2], uuid[2:4], uuid[4:], 'path', 'symbols.npy')
+            symbols = np.load(symbols_path).tolist()
+            os.remove(symbols_path)
+            # Update 'node_attributes'
+            data['node_attributes'][node_id].pop('array|symbols')
+            data['node_attributes'][node_id]['symbols'] = symbols
+            # Update 'node_attributes_conversion'
+            data['node_attributes_conversion'][node_id].pop('array|symbols')
+            data['node_attributes_conversion'][node_id]['symbols'] = [None] * len(symbols)
+
+
+def migration_remove_node_prefix(data):
+    """Apply migrations: 0028 - REV. 1.0.28
+    Change node type strings:
+    'node.data.' -> 'data.'
+    'node.process.' -> 'process.'
+    """
+    for value in data['export_data']['Node'].values():
+        if 'type' in value:
+            if value['type'].startswith('node.data.'):
+                value['type'] = value['type'].replace('node.data.', 'data.', 1)
+            elif value['type'].startswith('node.process.'):
+                value['type'] = value['type'].replace('node.process.', 'process.', 1)
+
+
+def migration_rename_parameter_data_to_dict(data):
+    """Apply migration: 0029 - REV. 1.0.29
+    Update ParameterData to Dict
+    """
+    for value in data['export_data']['Node'].values():
+        if 'type' in value and value['type'] == 'data.parameter.ParameterData.':
+            value['type'] = 'data.dict.Dict.'
+
+
+def migration_dbnode_type_to_dbnode_node_type(metadata, data):
+    """Apply migration: 0030 - REV. 1.0.30
+    Renaming DbNode.type to DbNode.node_type
+    """
+    # For data.json
+    if 'Node' in data['export_data']:
+        for content in data['export_data']['Node'].values():
+            if 'type' in content:
+                content['node_type'] = content.pop('type')
+    # For metadata.json
+    if 'type' in metadata['all_fields_info']['Node']:
+        metadata['all_fields_info']['Node']['node_type'] = metadata['all_fields_info']['Node'].pop('type')
+
+
+def migration_remove_dbcomputer_enabled(metadata, data):
+    """Apply migration: 0031 - REV. 1.0.31
+    Remove DbComputer.enabled
+    """
+    # For data.json
+    if 'Computer' in data['export_data']:
+        for content in data['export_data']['Computer'].values():
+            if 'enabled' in content:
```

Comment:
for reference: these two lines are typically written as `content.pop('enabled', None)` which means try to pop the key, but if it does not exist, no biggy

Reply:
done